Nginx update to ver 1.13.3

gaydabura opened the pull request.

gaydabura committed:
Nginx update to ver 1.13.3

gaydabura committed:
Nginx update to ver 1.13.3

gaydabura committed:
Nginx update to ver 1.13.3

gaydabura commented:
updated PR. please review it

andypost commented on Dockerfile:
```diff
@@ -20,6 +20,7 @@ RUN addgroup -S -g 82 www-data && \
 RUN apk add --update \
 
         # Base packages
+        bash \
```
please remove bash, alpine has busybox with `ash`

andypost commented on Dockerfile:
```diff
@@ -112,9 +109,19 @@ COPY nginx.conf /etc/nginx/nginx.conf
+RUN sed -i "s!{{ NGINX_DOCUMENT_ROOT }}!${NGINX_DOCUMENT_ROOT}!g" /etc/nginx/conf.d/drupal8.conf
+RUN sed -i "s/{{ NGINX_FASTCGI_BUFFERS }}/${NGINX_FASTCGI_BUFFERS}/" /etc/nginx/nginx.conf
+RUN sed -i "s/{{ NGINX_FASTCGI_BUFFER_SIZE }}/${NGINX_FASTCGI_BUFFER_SIZE}/" /etc/nginx/nginx.conf
+RUN sed -i "s/{{ NGINX_FASTCGI_READ_TIMEOUT }}/${NGINX_FASTCGI_READ_TIMEOUT}/" /etc/nginx/nginx.conf
```
all this RUNs should be oneliner to have less layers in image

andypost commented on Dockerfile:
```diff
@@ -112,9 +109,19 @@ COPY nginx.conf /etc/nginx/nginx.conf
+
+
+WORKDIR ${NGINX_DOCUMENT_ROOT}
+VOLUME ${NGINX_DOCUMENT_ROOT}
```
I see no reason to make it configurable with ENV because it will be set in stone in image anyway

andypost commented on drupal8.conf:
```diff
@@ -2,48 +2,46 @@ server {
     location / {
-
-        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
-        proxy_set_header X-Forwarded-Proto $scheme;
```
why that removed?

for every project we use following things to pass
```
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection 'upgrade';
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_cache_bypass $http_upgrade;
```

andypost commented on Dockerfile:
```diff
@@ -1,17 +1,17 @@
-ENV LUAJIT_LIB /usr/lib/
-ENV LUAJIT_INC /usr/include/luajit-2.0/
+ENV NGINX_DOCUMENT_ROOT /var/www/html/
+ENV NGINX_FASTCGI_BUFFERS "16 32k"
```
are you sure that 16 buffers enough?

gaydabura replied:
by default `8 4k` and its value you can change locally for your project

andypost reviewed (changes requested):
- You make too many layers in image
- remove bash
- no reason to configure volume&workdir with env
- add back proxy options

gaydabura committed:
Nginx update to ver 1.13.3

andypost commented on drupal8.conf:
```diff
@@ -235,4 +212,5 @@ server {
     }
-}
+
+}
```
(

andypost commented on Dockerfile:
```diff
@@ -112,9 +108,9 @@ COPY nginx.conf /etc/nginx/nginx.conf
-VOLUME /var/www/html
+RUN if [ -z $NGINX_FASTCGI_BUFFERS ] ; then export NGINX_FASTCGI_BUFFERS="16 32k"; fi; if [ -z $NGINX_FASTCGI_BUFFER_SIZE ] ; then export NGINX_FASTCGI_BUFFER_SIZE="32k"; fi; if [ -z $NGINX_FASTCGI_READ_TIMEOUT ] ; then export NGINX_FASTCGI_READ_TIMEOUT="900"; fi; if [ -z $NGINX_DOCUMENT_ROOT ] ; then export NGINX_DOCUMENT_ROOT="/var/www/html/"; fi
+RUN sed -i "s!{{ NGINX_DOCUMENT_ROOT }}!${NGINX_DOCUMENT_ROOT}!g" /etc/nginx/conf.d/drupal8.conf
+RUN sed -i "s/{{ NGINX_FASTCGI_BUFFERS }}/${NGINX_FASTCGI_BUFFERS}/;s/{{ NGINX_FASTCGI_BUFFER_SIZE }}/${NGINX_FASTCGI_BUFFER_SIZE}/;s/{{ NGINX_FASTCGI_READ_TIMEOUT }}/${NGINX_FASTCGI_READ_TIMEOUT}/" /etc/nginx/nginx.conf
```
still lots of layers(

andypost commented on README.md:
```diff
@@ -1 +1,10 @@
\ No newline at end of file
+# skilld-docker-nginx
+
+## Environment Variables
```
actually it is not true, because this vars applied only once at container build time

andypost reviewed (changes requested):
Still require to minimize number of layers inside image

alexkras committed:
Updated:
alpine - edge,
nginx - 1.13.8,
upload_progress - 0.9.0,
naxsi - 0.55.3
removed comments line inside RUN section removed check for ENV vars.

alexkras committed:
Switch nginx to latest stable 1.12.2

gaydabura committed:
Merge pull request #13 from alexkras/f1.13.3

Updated: latest nginx 1.13.8 based on f1.13.3 branch

gaydabura committed:
Revert "Updated: latest nginx 1.13.8 based on f1.13.3 branch"

gaydabura committed:
Merge pull request #15 from skilld-labs/revert-13-f1.13.3

Revert "Updated: latest nginx 1.13.8 based on f1.13.3 branch"

alexkras commented:
Need to review last changes

uzlov commented on drupal8.conf:
```diff
@@ -2,48 +2,56 @@ server {
     location / {
 
+        proxy_http_version 1.1;
+        proxy_set_header Upgrade $http_upgrade;
```
lines 17-23 for proxying of websockets

andypost commented:
I think it outdated enough
